scripts/editor/output-css-variables.js

```diff
@@ -72,6 +72,142 @@ export const globalInner = (itemValues, itemKey) => {
+ */
+const setupResponsiveVariables = (responsiveAttributes, variables) => {
+	// Iterate through responsive attributes.
+	return Object.entries(responsiveAttributes).reduce((responsiveAttributesVariables, [responsiveAttributeName, responsiveAttributeObject]) => {
+
+		// If responsive attribute doesn't exist in variables object, skip it
+		if (!responsiveAttributeName || _.isEmpty(variables[responsiveAttributeName])) {
+			return responsiveAttributesVariables;
+		}
+
+		// Used for determination of default breakpoint.
+		const numberOfBreakpoints = Object.entries(responsiveAttributeObject).length;
+
+		// Iterate each responsive attribute object as breakpoint name is the key of the object, and value represents the name of the responsive variable.
+		const responsiveAttributeVariables = Object.entries(responsiveAttributeObject).reduce((responsiveAttribute, [breakpointName, breakpointVariableName], breakpointIndex) => {
+			let breakpointVariables = {};
+
+			// Array represents direct value(default or value).
+			if (Array.isArray(variables[responsiveAttributeName])) {
+				breakpointVariables = setBreakpointResponsiveVariables(variables[responsiveAttributeName], breakpointName, breakpointIndex, numberOfBreakpoints);
+
+			// Object treatment goes depending on a value inserted(multiple choice, boolean or similar).
+			} else {
+
+				// Iterate options/multiple choices/boolean...
+				breakpointVariables = Object.entries(variables[responsiveAttributeName])
+				.reduce((acc, [attributeValue, attributeObject]) => {
+					return {
+						...acc,
+						[attributeValue]: setBreakpointResponsiveVariables(attributeObject, breakpointName, breakpointIndex, numberOfBreakpoints),
+					}
+				}, {});
+			}
+
+			// Collect all the values from one responsive attribute to one object.
+			return {
+				...responsiveAttribute,
+				[breakpointVariableName]: breakpointVariables,
+			};
+		}, {});
+
+		// Merge multiple responsive attributes to one object.
+		return {...responsiveAttributesVariables, ...responsiveAttributeVariables};
+	}, {});
+}
+
+/**
+ * Setting defined variables to each breakpoint.
+ *
+ * @param {object} attributes - Attributes fetched from manifest.
+ * @param {object} variables  - Variables fetched from manifest.
+ * @param {object} data       - Preset objects separated in breakpoints.
+ *
+ * @return {object} Filled object with variables data separated in breakpoints.
+ */
+const setVariablesToBreakpoints = (attributes, variables, data) => {
+	// Iterate each variable.
+	for (const [variableName, variableValue] of Object.entries(variables)) {
+
+		// Constant for attributes set value (in db or default).
+		const attributeValue = attributes[variableName];
+
+		// Set internal breakpoints variable.
+		let internalBreakpoints = [];
+
+		// If type default or value.
+		if (!Array.isArray(variableValue)) {
+			internalBreakpoints = variableValue[attributeValue] ?? [];
+		} else {
+			internalBreakpoints = variableValue;
+		}
```

Comment:
Unnecessary `else`. You could just set the `internalBreakpoints` to `variableValue` and if the `variableValue` is an array, set it to the value inside the `if`.

Reply:
Would `const internalBreakpoints =  Array.isArray(variableValue) ? variableValue : (variableValue[attributeValue] ?? []);`
be a better solution?
(to be honest, I didn't want to change it I just extracted the function)

I don't like that we are overriding variables at any point because any kind of modification might influence the outcome. And in this case, there is no need to make it anything other than constant. Would you agree?

Reply:
What I meant is something like this:

```js
let internalBreakpoints = variableValue;

// If type default or value.
if (!Array.isArray(variableValue)) {
	internalBreakpoints = variableValue[attributeValue] ?? [];
}
```

That way the `internalBreakpoints` variable will always be set to something, but overwritten to `variableValue[attributeValue] ?? [];` if it's not an array.

> I don't like that we are overriding variables at any point because any kind of modification might influence the outcome. And in this case, there is no need to make it anything other than constant. Would you agree?

Not sure I understand the first part. You are already overriding the `internalBreakpoints` (you've initialized it to be an empty array).

Reply:
> Not sure I understand the first part. You are already overriding the internalBreakpoints (you've initialized it to be an empty array).

That's why I suggested
` const internalBreakpoints = Array.isArray(variableValue) ? variableValue : (variableValue[attributeValue] || []);`

The current
```
if (!Array.isArray(variableValue)) {
			internalBreakpoints = variableValue[attributeValue] ?? [];
		} else {
			internalBreakpoints = variableValue;
		}
```
was not written by me. This whole function was just extracted from the code so I can reuse it.

Reply:
Karlo's last suggestion is fine, although I definitely find Denis' version cleaner (initialized to value + if) to read.

BTW Karlo if you go with your last suggested ternary, make sure to change the _or_ operator to _nullish coalescing_
```js
const internalBreakpoints = 
    Array.isArray(variableValue) 
    ? variableValue 
    : (variableValue[attributeValue] || []);
```
to
```js
const internalBreakpoints = 
    Array.isArray(variableValue) 
    ? variableValue 
    : (variableValue[attributeValue] ?? []);
```

Reply:
You can even refactor to what Goc suggested, looks easier to read. Althougth I leave the ternary characters in the same line

```js
const internalBreakpoints = 
    Array.isArray(variableValue) ?
    variableValue :
    (variableValue[attributeValue] ?? []);
```

You decide which one looks best for you @volfkarlo 😄